pyaedt/edb_grpc/core/edb_data/hfss_simulation_setup_data.py

```diff
@@ -0,0 +1,2146 @@
+        """Maximum change of S-parameters between two consecutive passes, which serves as
+        a stopping criterion.
+
+        Returns
+        -------
+        str
+        """
+        return self._adaptive_frequency_data.MaxDelta
+
+    @max_delta.setter
+    def max_delta(self, value):
+        self._adaptive_frequency_data.MaxDelta = str(value)
+
+    @property
+    def max_passes(self):
+        """Maximum allowed number of mesh refinement cycles.
+
+        Returns
+        -------
+        int
+        """
+        return self._adaptive_frequency_data.MaxPasses
+
+    @max_passes.setter
+    def max_passes(self, value):
+        self._adaptive_frequency_data.MaxPasses = value
+
+
+class AdaptiveSettings(object):
+    """Manages EDB methods for adaptive settings."""
+
+    def __init__(self, parent):
+        self._parent = parent
+        self._adapt_type_mapping = {
+            "kSingle": self.adaptive_settings.AdaptType.kSingle,
+            "kMultiFrequencies": self.adaptive_settings.AdaptType.kMultiFrequencies,
+            "kBroadband": self.adaptive_settings.AdaptType.kBroadband,
+            "kNumAdaptTypes": self.adaptive_settings.AdaptType.kNumAdaptTypes,
+        }
+
+    @property
+    def adaptive_settings(self):
+        """Adaptive EDB settings.
+
+        Returns
+        -------
+        :class:`pyaedt.edb_grpc.core.edb_data.hfss_simulation_setup_data.AdaptiveSettings`
+        """
+        return self._parent._edb_sim_setup_info.SimulationSettings.AdaptiveSettings
+
+    @property
+    def adaptive_frequency_data_list(self):
+        """List of all adaptive frequency data.
+
+        Returns
+        -------
+        :class:`pyaedt.edb_grpc.core.edb_data.hfss_simulation_setup_data.AdaptiveFrequencyData`
+        """
+        return [AdaptiveFrequencyData(i) for i in list(self.adaptive_settings.AdaptiveFrequencyDataList)]
+
+    @property
+    def adapt_type(self):
+        """Adaptive type.
+        Options:
+        1- ``kSingle``.
+        2- ``kMultiFrequencies``.
+        3- ``kBroadband``.
+        4- ``kNumAdaptTypes``.
+
+        Returns
+        -------
+        str
+        """
+        return self.adaptive_settings.AdaptType.ToString()
+
+    @adapt_type.setter
+    def adapt_type(self, value):
+        self.adaptive_settings.AdaptType = self._adapt_type_mapping[value]
+        self._parent._update_setup()
+
+    @property
+    def basic(self):
+        """Whether if turn on basic adaptive.
+
+        Returns
+        -------
+            ``True`` if bais adaptive is used, ``False`` otherwise.
+        """
+        return self.adaptive_settings.Basic
+
+    @basic.setter
+    def basic(self, value):
+        self.adaptive_settings.Basic = value
+        self._parent._update_setup()
+
+    @property
+    def do_adaptive(self):
+        """Whether if adpative mesh is on.
+
+        Returns
+        -------
+        bool
+            ``True`` if adaptive is used, ``False`` otherwise.
```

Comment:
```suggestion
        """Whether adaptive mesh is used.

        Returns
        -------
        bool
            ``True`` if adaptive is used, ``False`` otherwise.
```